scripts/west_commands/runners/bossac.py

```diff
@@ -56,8 +56,9 @@ def do_run(self, command, **kwargs):
                     'ospeed', '1200', 'cs8', '-cstopb', 'ignpar', 'eol', '255',
                     'eof', '255']
         cmd_flash = [self.bossac, '-p', self.port, '-R', '-e', '-w', '-v',
-                     '-o', '%s' % self.offset,
                      '-b', self.cfg.bin_file]
+        if self.offset is not None:
+            cmd_flash = cmd_flash + ['-o', '%s' % self.offset]
```

Comment:
Minor nit, an in-place update would be slightly preferable style-wise:

```suggestion
            cmd_flash += ['-o', '%s' % self.offset]
```